drivers/hmis/app/graphql/types/hmis_schema/occurrence_point_form.rb

```diff
@@ -14,22 +14,17 @@ class HmisSchema::OccurrencePointForm < Types::BaseObject
     # Form used for Viewing/Creating/Editing records
     field :definition, Types::Forms::FormDefinition, null: false, extras: [:parent]
 
-    # object is a Hmis::Form::Instance
+    # object is an OpenStruct, see Hmis::Hud::Enrollment occurrence_point_forms
 
     def id(parent:)
       # Include project id (if present) so that instance is not cached for use across projects.
-      [object.id, parent&.id].compact.join(':')
-    end
-
-    def data_collected_about
-      object.data_collected_about || 'ALL_CLIENTS'
+      [object[:id], parent&.id].compact.join(':')
```

Comment:
So the ID was previously  `<instance id>:<project id>` is now `<definition id>:<project id>` is that right?

Maybe it would be a little clearer if we just choose the `id` here within the Type, instead of each place where we create the OpenStruct. That way it's easier to tell what the `id` actually contains. For example:


When creating Open Structs:
```ruby
        OpenStruct.new(
          # id: instance.definition.id, <= removed this
          legacy: false,
          definition: instance.definition,
          data_collected_about: instance.data_collected_about || 'ALL_CLIENTS',
        )
```

And when resolving type:
```ruby
    def id(parent:)
      # Include project id (if present) so that instance is not cached for use across projects.
      [object.definition.id, parent&.id].compact.join(':')
    end
```

Does that make sense?

Reply:
Yes that makes sense to me. The parent can be either an enrollment or a project now, so I updated to account for that.

Reply:
Ok I see. thank you for your flexibility with new changes and requirements popping up!